packages/lesswrong/components/ea-forum/TargetedJobAdSection.tsx

```diff
@@ -73,53 +83,103 @@ const TargetedJobAdSection = () => {
-      // check if the ad fits the user's interests -
-      // currently based on whether they have subscribed to all the topics relevant to the job ad
-      const subscribedTagIds = JOB_AD_DATA[jobName].subscribedTagIds
+      const userJobAdState = ads.find(ad => ad.jobName === jobName)?.adState
+      const userEAGData = userEAGDetails?.[0]
+      
+      /** Check if the job fits the user's interests */
+
+      // Are they subscribed to all the topics relevant to the job ad?
+      const subscribedTagIds = jobAd.subscribedTagIds
       const userTagSubs = currentUser.frontpageFilterSettings?.tags?.filter(
         (setting: FilterTag) => filterModeIsSubscribed(setting.filterMode)
       )?.map((setting: FilterTag) => setting.tagId)
       let userIsMatch = subscribedTagIds && !difference(subscribedTagIds, userTagSubs).length
-      // or if they have read at least 30 posts in all the relevant topics in the past 6 months
-      const readsThreshold = JOB_AD_DATA[jobName].coreTagReadsThreshold ?? 30
+      // Or have they have read at least 30 posts in all the relevant topics in the past 6 months?
+      const readCoreTagIds = jobAd.readCoreTagIds
+      const readsThreshold = jobAd.coreTagReadsThreshold ?? 30
       userIsMatch = userIsMatch || (
+        readCoreTagIds &&
         !!coreTagReads?.length &&
-        JOB_AD_DATA[jobName].readCoreTagIds?.every(
+        readCoreTagIds.every(
           tagId => coreTagReads.some(tag => tag.tagId === tagId && tag.userReadCount >= readsThreshold)
         )
       )
+      // Or do they have the relevant EAG interests?
+      const interestedIn = jobAd.interestedIn
+      const userEAGInterests = union(userEAGData?.interestedIn, userEAGData?.experiencedIn)
+      userIsMatch = userIsMatch || (
+        !!interestedIn &&
+        !!userEAGInterests &&
+        !difference(interestedIn, userEAGInterests).length
+      )
+      // In the rare case when we don't want to filter by interests, start by matching on all users
+      if (!subscribedTagIds && !readCoreTagIds && !interestedIn) {
+        userIsMatch = true
+      }
       
-      // check if the ad is limited to a certain country
-      const countryCode = JOB_AD_DATA[jobName].countryCode
-      if (userIsMatch && countryCode) {
-        userIsMatch = getCountryCode(currentUser.googleLocation) === countryCode
+      /** Check if the user should be excluded from the job ad audience */
+
+      // Are they in the right country?
+      const countryCode = jobAd.countryCode
+      const countryName = jobAd.countryName
+      if (userIsMatch && (countryCode || countryName)) {
+        userIsMatch = (getCountryCode(currentUser.googleLocation) === countryCode) ||
+          (userEAGData?.countryOrRegion === countryName)
+      }
+      // And are they near the right city?
+      const nearestCity = jobAd.nearestCity
+      if (userIsMatch && nearestCity) {
+        userIsMatch = userEAGData?.nearestCity === nearestCity
```

Comment:
This one seems a bit fragile, there are cities that are very close to each other (SF/Berkeley and Boston/Cambridge are important ones). Also this filters them out if they haven't filled in their city which doesn't seem ideal, I think it would be better to just fall back to the country in that case.

Edit: there is also a willingnessToRelocate field which would be good to include as well

Reply:
My feeling with all of these filters is that they are on the overly strict side, especially when combined. I think it would be good to run a query to check how many users actually match each job, and potentially loosen these if it's not very many

Reply:
Yeah I added some stuff while just playing around with various jobs. The city one isn't used now so I can remove it.

These filters are meant to be somewhat stricter than before - is there something specific you think should be changed? I do spend time on each job I consider messing with the filters, you can see my notes for the current round of jobs in [this doc](https://docs.google.com/document/d/1FEa4GHQSetBll-ex3aQogF3bJSgWrfo5t2mrJBkyZHk/edit#heading=h.pwtkkxql4027). I lean towards being stricter because of the prominence of this UI element, and because we don't want any individual users to see it too often. I'd be quite happy to hear your thoughts on my approach!